public/static/app/perf/perf.js

```diff
@@ -631,27 +658,36 @@ mciModule.controller('PerfController', function PerfController(
         // Populate the trend data
         var chartDataQ = $http.get("/plugin/json/history/" + $scope.task.id + "/perf").then(
           function(resp) {
-            $scope.trendSamples = new TrendSamples(resp.data);
-            $scope.metricSelect.options = [$scope.metricSelect.default].concat(
-              _.map(
-                _.without($scope.trendSamples.metrics, $scope.metricSelect.default.key),
-                function(d) { return {key: d, name: d} }
+            getRejectedPointsQ.then(function(rejects){
+              let data = resp.data;
+              if(rejects.length){
+                data = _.reject(resp.data, function(doc) {
+                  // return  $scope.task.id != doc.task_id && _.contains(rejects, doc.task_id);
+                  return  _.contains(rejects, doc.task_id);
+                })
+              }
+              $scope.trendSamples = new TrendSamples(data);
+              $scope.metricSelect.options = [$scope.metricSelect.default].concat(
+                _.map(
+                  _.without($scope.trendSamples.metrics, $scope.metricSelect.default.key),
+                  function(d) { return {key: d, name: d} }
+                )
               )
-            )
-
-          // Some copypasted checks
-          if ($scope.conf.enabled){
-            if ($location.hash().length > 0) {
-              try {
-                if ('metric' in hashparsed) {
-                  let metric = hashparsed.metric
-                  $scope.metricSelect.value = _.findWhere(
-                    $scope.metricSelect.options, {key: metric}
-                  ) || $scope.metricSelect.default
+
+              // Some copypasted checks
+              if ($scope.conf.enabled){
+                if ($location.hash().length > 0) {
+                  try {
+                    if ('metric' in hashparsed) {
+                      let metric = hashparsed.metric
```

Comment:
[nit] Could this 'let' be a 'const'? It doesn't looks like the value is ever changed.

Reply:
Its simpler just to remove the decl altogether?

Reply:
That is fine with me. And I realize this code was there before. I just always default to 'const' unless I have to be less strict.